packages/plugin-ext/src/common/plugin-api-rpc.ts

```diff
@@ -162,12 +152,35 @@ export const emptyPlugin: Plugin = {
     }
 };
 
+export interface PluginManagerInitializeParams {
```

Comment:
I don't see why there is a rename from PluginInitData to PluginManagerInitializeParams. I mean it's already a lot of changes so mixing some opinionated cleanup and changes is something that I dislike as I already commented in a lot of PRs

Reply:
We used to send data to initialize manager each time when we start a plugin, i.e. all preferences. I've changed it to send them only once when the manager is created, so we don't need to push them again and again on each deployment.

Reply:
but I mean, you could just change params.

It's like renaming init to initialize... why ?

Reply:
ok, just to check that I understand correctly:
- we keep `$init` but we call it only once without options to start plugins
- we add `$start` method to actually start plugins

Reply:
I mean it's a global thing: please separate cleanup/code refactoring from enhancement. Here we have both mixed

cleanup, code refactoring is good but for me mixing it with enhancements is producing nightmare and unexpected results in downstream as well

Reply:
> I mean it's a global thing: please separate cleanup/code refactoring from enhancement. Here we have both mixed

@benoitf I agree that we should avoid doing "opinionated" improvements, i.e. changes based on taste or generally without a clear benefit. But generally improving the code is a good thing, no? I don't see why improvements should not go together with bugfixes and enhancements. I'd actually like to encourage everyone to strive for improving the code they touch. Maybe I don't understand your concern and we can discuss in next dev meeting?

Reply:
@svenefftinge my remark is just to separate commits from what is feature and what is cleanup/refactoring and not do enhancement, and everything as well in the same time

I usually see ppl changing code formatting, renaming methods, etc as part of a new feature commit and I dislike it personally as from reviewer, you want to see if the change is really for the feature or not

Reply:
in short, small commits, small PRs, merge fast

Reply:
this PR is doing code cleanup, introduce breaking changes, remove partially the raw model with hack,  enable perMessage deflate compression, change the way metadata are kept, etc. but I would say almost all of them could be separated and merged in independent way. break less, easy to track, easy to revert

Reply:
commit history is rewritten with a commit for each improvement, is there something else have to be done? Please have a look at other comments whether they are resolved that we can proceed.